test/support/utils.rb

```diff
@@ -47,41 +48,44 @@ def create_pseudo_terminal(boot_options: "-r debug/run")
 
       ENV['RUBYOPT'] = "-I #{lib}"
       ENV['RUBY_DEBUG_USE_COLORIZE'] = "false"
+      ENV['RUBY_DEBUG_TEST_MODE'] = 'true'
 
       PTY.spawn("#{RUBY} #{boot_options} #{temp_file_path}") do |read, write, pid|
-        quit = false
-        result = nil
-
-        until quit
-          read.expect(/(.*)\n|\(rdbg\)/) do |sentence|
-            debug_print sentence[0]
-            if sentence[0] == '(rdbg)'
-              cmd = @queue.pop
-              if cmd.is_a?(Proc)
-                cmd.call(result)
+        lines = []
+        begin
+          Timeout.timeout(1) do
```

Comment:
How to determine "1 second"?

Reply:
@ko1 
I think 1 second is enough time. What do you think about it?

Reply:
Good idea to ask it :)

For me, 1 second is too short. If it runs complex tests, they will take more time. Also on the low-power computers, this  limitation should be more problematic. Tests is expected to finish sometime, so you need to check the "unexpected freeze". In many cases maybe 10 seconds are fine.

Introducing a way to change this timeout will help for slow computers (RUBY_DEBUG_TIMEOUT_SCALE for example), but now it is too much.

Reply:
Thanks, I will change it!